packages/manager/modules/pci/src/projects/project/storages/volume-backup/list/restore-volume/controller.js

```diff
@@ -0,0 +1,58 @@
+export default class VolumeBackupRestoreVolumeController {
+  /* @ngInject */
+  constructor($translate, VolumeBackupService) {
+    this.$translate = $translate;
+    this.volumeBackupService = VolumeBackupService;
+  }
+
+  $onInit() {
+    this.isRestoring = false;
+  }
+
+  onRestoreVolumeClick() {
+    // TODO: Tracking -- MANAGER-10570
+
+    this.isRestoring = true;
+    return this.volumeBackupService
+      .restoreVolumeBackupToVolume(
+        this.projectId,
+        this.volumeBackup.region,
+        this.volumeBackup.id,
+        this.volume.id,
+      )
+      .then(() => {
+        return this.goToVolumeBackups({
+          textHtml: this.$translate.instant(
+            'pci_projects_project_storages_volume_backup_list_restore_volume_action_request_success',
+            {
+              volumeName: `<strong>${this.volume.name}</strong>`,
+            },
+          ),
+        });
+      })
+      .then(() => this.startPolling())
```

Comment:
**nitpick:** Calling a `startPolling` after the state navigation affects the destroy of this controller. Please poll in the destination state controller.

Reply:
Hello 🤝 Anoop, i guess no, because the startPolling function is part of root volume-backup.
So, it will not prevent to close properly the restore volume state.

Reply:
As discussed together, the modal is not preventing to excute the Angular `$onDestroy` hook.
![image](https://user-images.githubusercontent.com/22635766/222686830-8e04b62e-bfd6-4179-96d2-61c80eb07bba.png)